library/asn1write.c

```diff
@@ -287,26 +287,73 @@ int mbedtls_asn1_write_ia5_string( unsigned char **p, unsigned char *start,
     return( (int) len );
 }
 
+int mbedtls_asn1_write_named_bitstring( unsigned char **p,
+                                        unsigned char *start,
+                                        const unsigned char *buf,
+                                        size_t bits )
+{
+    size_t unused_bits, byte_len;
+    unsigned char cur_byte;
+
+    /*
+     * Named bitstrings require that trailing 0s are excluded in the encoding
+     * of the bitstring. Trailing 0s are considered part of the 'unused' bits
+     * when encoding this value in the first content octet
+     */
+    byte_len = ( bits / 8 ) + ( ( bits % 8 ) ? 1 : 0 );
+    unused_bits = ( 8 - ( bits % 8 ) ) % 8;
+
+    /* Scan the input data from back to front and remove 0-value bytes */
+    for( ; byte_len > 0; byte_len-- )
+    {
+        cur_byte = ( buf[byte_len - 1] >> unused_bits ) << unused_bits;
```

Comment:
Minor: The final left shift is unnecessary when it comes to checking whether `cur_byte == 0`.